src/lib/Pagination/Pagerfanta/TrashItemAdapter.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\EzPlatformAdminUi\Pagination\Pagerfanta;
+
+use eZ\Publish\API\Repository\TrashService;
+use eZ\Publish\API\Repository\Values\Content\Query;
+use Pagerfanta\Adapter\AdapterInterface;
+
+/**
+ * Pagerfanta adapter for eZ Publish content search.
+ * Will return results as SearchHit objects.
+ */
+class TrashItemAdapter implements AdapterInterface
+{
+    /**
+     * @var \eZ\Publish\API\Repository\Values\Content\Query
+     */
+    private $query;
+
+    /**
+     * @var \eZ\Publish\API\Repository\TrashService
+     */
+    private $trashService;
+
+    /**
+     * @var int
+     */
+    private $nbResults;
+
+    public function __construct(Query $query, TrashService $trashService)
+    {
+        $this->query = $query;
+        $this->trashService = $trashService;
+    }
+
+    /**
+     * Returns the number of results.
+     *
+     * @return int the number of results
+     */
+    public function getNbResults()
+    {
+        if (isset($this->nbResults)) {
+            return $this->nbResults;
+        }
+
+        $countQuery = clone $this->query;
+        $countQuery->limit = 0;
+
+        return $this->nbResults = $this->trashService->findTrashItems($countQuery)->count;
```

Comment:
Hmm, this probably needs PAPI improvement, because in case of 10000 items we still need to fetch them all to get count.
// cc @andrerom 

@mikadamczyk question for now: is it possible to use pagination w/o information about count? Yeah, that's kinda silly, but maybe...?

Reply:
@alongosz Given what is available, API use seems okish, and we can with not to much effort do work under the hood to across 6.x and 7.x to improve it to just do a count in this case.

That said the api end point should ideally be either deprecated in favor of a much simpler one, or we need to model some kind of query that should be supported for trash, as the query itself is ignored for trash atm.